plugins/hotkeyshortcuts.koplugin/main.lua

```diff
@@ -0,0 +1,581 @@
+        end,
+    }
+    menu_items.hotkeyshortcuts = {
+        sorting_hint = "physical_buttons_setup",
+        text = _("Keyboard shortcuts"),
+        sub_item_table = {
+            {
+                text = _("Cursor keys"),
+                sub_item_table = self:genSubItemTable(cursor_keys),
+            },
+            {
+                text = _("Page-turn buttons"),
+                enabled_func = function() return self.hotkey_mode == "hotkeyshortcuts_reader" end,
+                sub_item_table = self:genSubItemTable(pg_turn),
+            },
+            {
+                text = _("Function keys"),
+                sub_item_table = self:genSubItemTable(fn_keys),
+            },
+        },
+    }
+
+    if Device:hasKeyboard() then
+        table.insert(menu_items.hotkeyshortcuts.sub_item_table, {
+            text = _("Alphabet keys"),
+            sub_item_table = self:genSubItemTable({
+                "alt_plus_a", "alt_plus_b", "alt_plus_c", "alt_plus_d", "alt_plus_e", "alt_plus_f", "alt_plus_g", "alt_plus_h", "alt_plus_i",
+                "alt_plus_j", "alt_plus_k", "alt_plus_l", "alt_plus_m", "alt_plus_n", "alt_plus_o", "alt_plus_p", "alt_plus_q", "alt_plus_r",
+                "alt_plus_s", "alt_plus_t", "alt_plus_u", "alt_plus_v", "alt_plus_w", "alt_plus_x", "alt_plus_y", "alt_plus_z",
+            }),
+        })
+    end
+end
+
+--[[
+    Description:
+    This function overrides existing `registerKeyEvents()` functions in various modules to resolve conflicts and customize key event handling.
+    It modifies the key event registration for several reader and file manager modules based on device capabilities and user settings.
+
+    Modules and their modifications:
+    - ReaderBookmark: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderConfig: Customizes `ShowConfigMenu` key event based on user settings.
+    - ReaderDictionary: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderLink: Customizes `GotoSelectedPageLink` key event for devices with screen keyboard or symbol key.
+    - ReaderSearch: Customizes `ShowFulltextSearchInputBlank` key event for devices with a keyboard.
+    - ReaderToc: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderThumbnail: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderWikipedia: Overrides `registerKeyEvents()` with an empty function.
+    - ReaderUI: Customizes `Home` and `KeyContentSelection` key events based on device capabilities.
+    - FileManager: Customizes `Home` and `Back` key events, and conditionally removes `Close` key event.
+    - FileSearcher: CCustomizes `ShowFileSearchBlank` key event for devices with keyboards.
+    - FileManagerMenu: Customizes `ShowMenu` key event for devices with keys.
+]]
+function HotKeyShortcuts:overrideConflictingFunctions()
+    local ReaderBookmark = require("apps/reader/modules/readerbookmark")
+    ReaderBookmark.registerKeyEvents = function(readerbookmark)
+    end
```

Comment:
overrideConflictingFunctions() seems to be called always when the plugin is enabled ?
And when the user is not interested by this plugin, you will just remove the original key registration, and plug back an _equivalent/identical_ mapping, right ?
Feels a bit strange, but probably easier than detecting if needed and restoring when the assignment gets back to be similar to the defaults. So... ok.

Anyway, would read simpler with these on a single line when the func is a no-op:
`ReaderBookmark.registerKeyEvents = function(readerbookmark) end`

Reply:
it is called as part of `registerKeyEvents()` during init so it is only enabled if the plugin is actually active. If someone decides to have the plugin off, it reverts back to core's `keyEvents()`